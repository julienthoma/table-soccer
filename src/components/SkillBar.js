import React from 'react';
import PropTypes from 'prop-types';
import './SkillBar.scss';
import { getFormattedPercent } from '../services/formatter';

const SkillBar = ({
  factor,
  leftHeadline,
  value,
  best,
  bestPlayerId
}) =>
  <div styleName="root">
    <div styleName="headlineContainer">
      <h4 styleName="leftHeadline">{leftHeadline}</h4>
      <h4 styleName="rightHeadline">Best</h4>
    </div>

    <div styleName="bar">
      <div styleName="left" style={{ width: getFormattedPercent(factor) }} />
      <div styleName="leftText">{value}</div>
      <div styleName="right" />
      <div styleName="rightText">
        {best} ({bestPlayerId})
      </div>
    </div>
  </div>;

SkillBar.propTypes = {
  factor: PropTypes.number.isRequired,
  leftHeadline: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  best: PropTypes.string.isRequired,
  bestPlayerId: PropTypes.string.isRequired
};

export default SkillBar;
