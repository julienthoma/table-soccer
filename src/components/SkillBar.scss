@import '../scss/colors.scss';

.root {
  width: 100%;
  padding: 5px 0;
  overflow: hidden;
}

.headlineContainer {
  display: flex;
  justify-content: space-between;
}

.leftHeadline, .rightHeadline {
  margin: 0;
  font-size: 15px;
  color: #6d6d6d;
  font-weight: 400;
}

.headlineLeft {
  margin: 0;
}

.headlineRight {
  margin: 0;
}

.bar {
  position: relative;
  display: flex;
  width: 100%;
}

.left {
  position: relative;
  background: $blue;
  height: 20px;
}

.right {
  background: #dddddd;
  height: 20px;
  flex-grow: 1;
}

.leftText, .rightText, .valueText {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 12px;
}

.leftText {
  left: 10px;
}

.rightText {
  right: 10px;
}
