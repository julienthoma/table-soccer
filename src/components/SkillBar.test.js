import React from 'react';
import { renderWithMuiTheme } from '../test.utils';
import SkillBar from './SkillBar';

it('SkillBar renders correctly', () => {
  const bar = (
    <SkillBar
      leftHeadline="Hello"
      factor={0.3}
      value={20}
      best={100}
      bestPlayerId="fubar"
    />
  );
  expect(renderWithMuiTheme(bar)).toMatchSnapshot();
});
