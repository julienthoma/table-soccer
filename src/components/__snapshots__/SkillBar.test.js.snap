// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillBar renders correctly 1`] = `
.c2 {
  margin: 0;
  font-size: 15px;
  color: #6d6d6d;
  font-weight: 400;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c4 {
  position: relative;
  background: #00bcd4;
  height: 20px;
  width: 30%;
}

.c6 {
  position: relative;
  background: #dddddd;
  height: 20px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  left: 10px;
}

.c7 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  right: 10px;
}

.c0 {
  width: 100%;
  padding: 5px 0;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      Hello
    </h4>
    <h4
      className="c2"
    >
      Best
    </h4>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
      color="#00bcd4"
      width="30%"
    />
    <div
      className="c5"
    >
      20
    </div>
    <div
      className="c6"
      color="#dddddd"
    />
    <div
      className="c7"
    >
      100 (fubar)
    </div>
  </div>
</div>
`;
